src/acc/libsmm_acc/CMakeLists.txt

```diff
@@ -48,27 +48,27 @@ else () # i.e. USE_HIP
 
   set_source_files_properties(${LIBSMM_ACC_SRC_FILES}
                               PROPERTIES HIP_SOURCE_PROPERTY_FORMAT 1)
-  # hip_add_library does not support OBJECT libraries. For this reason, we are
-  # forced to use a static library
-  hip_add_library(libsmm_acc STATIC ${LIBSMM_ACC_FILES})
-  target_include_directories(libsmm_acc PRIVATE ${HIP_PATH}/../include)
+
+  set(HIP_HCC_FLAGS ${HIP_HCC_FLAGS} -fno-gpu-rdc)
+  hip_add_library(libsmm_acc SHARED ${LIBSMM_ACC_FILES})
   target_link_libraries(libsmm_acc INTERFACE "stdc++")
   target_link_libraries(libsmm_acc PUBLIC roc::hipblas)
+
   if (USE_OPENMP)
-    # since HIP is based on clang, not GCC, doing:
-    # target_link_libraries(libsmm_acc PUBLIC OpenMP::OpenMP_CXX) does not work
-    # - it links to GNU's OpenMP (libgomp.so) and not clang's libomp.so
-    target_link_libraries(libsmm_acc PUBLIC HIP_OpenMP)
-  endif ()
+    target_link_libraries(libsmm_acc PUBLIC OpenMP::OpenMP_CXX)
+  endif()
+
+  set_property(TARGET libsmm_acc PROPERTY POSITION_INDEPENDENT_CODE ON)
```

Comment:
does `hip_add_library` really not automatically set this for a `SHARED` lib?

Reply:
at least for CMake it is the default for a shared lib: https://cmake.org/cmake/help/latest/prop_tgt/POSITION_INDEPENDENT_CODE.html

Reply:
Let me try